infra/base-images/base-builder/compile_honggfuzz

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -eu
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+echo -n "Compiling honggfuzz to $LIB_FUZZING_ENGINE ..."
+
+pushd $SRC/honggfuzz > /dev/null
+make clean
+# Compile honggfuzz with gcc, as clang will be source of problems
```

Comment:
It does, but I'm using interesting constructs like

https://github.com/google/honggfuzz/blob/master/libcommon/common.h#L49

i.e. go's defer{} in C:). With gcc it works out-of-the-box. With clang it requires -fblocks and -lBlocksRuntime. I can add/require when compiling honggfuzz, but consequently the libhfuzz.a will also require it, and also all programs which are linking with it (fuzzed targers). To avoid the problem I'm compiling honggfuzz and libhfuzz.a with gcc, so it doesn't require additional options (bo be extact -lBlocksRuntime) from implementers of fuzzers.

It can be done, it's just with gcc it's simpler. But, of course, fuzzed libraries can be (and are) compiled with clang.